### DISCUSSION TITLE
[Feature request] Webhook for Dashboard Save

### DISCUSSION THREAD

**Discussion by siegenthalerroger** [grafana/grafana#37917] [General]
Adapted/Transfered from https://github.com/grafana/grafana/issues/9066

I have a usecase where some dashboards are loaded from provisioning and others are stored in DB. This is done so that our company can at least partly ensure that we have all relevant configuration coming from VCS while balancing the need for changes being made in the UI. A large issue in our setup is that if someone changes a Dashboard (creates a copy, or otherwise), there is no automated way of being informed of this happening, and as such no way for a process to be started to realign these changes with the VCS.

It would be of great help if there was a webhook that was triggered on every save of a dashboard. We'd be using this to create a Gitlab Issue, allowing us to kick of the VCS-checkin process.

**Comment by pkolyvas** [comment]
I wonder how close or far this is to the prototype functionality we demo'd at GrafanaCon this summer in the As-Code, simplified presentation here: https://grafana.com/go/grafanaconline/2021/dashboards-as-code/ (you'll need to provide an email for it to work.) The section from about 2m45s to 10m might be of particular interest. 

We're currently considering adding some subset of the functionality @malcolmholmes is demo'ing here (here = that video) with the [Grizzly](https://github.com/grafana/grizzly) tool directly into Grafana and I wonder if that would get us closer to your needs as well.

**Reply by malcolmholmes** [reply]
To get my GitHub/Git functionality to work, we first need to settle down on a workable API within Grafana itself. At present, when a dashboard is saved, a websocket event is published to other Grafana instances with the same dashboard open. I listen to that, pretending to be a browser. But that's a hacky way to do it. We need a more reliable and long term supportable way than this. @ryantxu

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
